perl/toolbox/json.pm

```diff
@@ -11,21 +11,60 @@ use IO::Uncompress::UnXz;
 use toolbox::logging;
 
 use Exporter qw(import);
-our @EXPORT = qw(put_json_file get_json_file);
+our @EXPORT = qw(put_json_file get_json_file open_write_text_file open_read_text_file);
 
 use strict;
 use warnings;
 
-sub put_json_file {
+
+sub open_write_text_file {
     my $filename = shift;
     chomp $filename;
-    my $json_ref = shift;
+    if (! defined $filename) {
+        printf "open_write_text_file(): filename was not defined\n";
+        exit 1;
+    }
+    if (! $filename =~ /\.xz/) { # Always default to compression when writing
+        $filename .= ".xz";
+    }
+    debug_log(sprintf "trying to open [%s] for writing\n", $filename);
+    my $fh = new IO::Compress::Xz $filename || die "Could not open file " . $filename;
```

Comment:
This should have the same check on the resulting file handle that `open_read_text_file` has.

Reply:
> I wonder if the `open_write_text_file` and `open_read_text_file` routines should return errors instead of exiting. As currently written, the caller can't do anything to handle these errors -- which may or may not be a problem. Thoughts?

I had them exit out because I think in all instances of using this the callers would exit anyway.  Also returning two vars IMO is kind of ugly in perl.

> This should have the same check on the resulting file handle that open_read_text_file has.

The only reason I did not do this was that I think the problem was with only reading existing files.  However, I can add it just in case.

Reply:
using return codes now

Reply:
added